core/src/Cuda/Kokkos_Cuda_Vectorization.hpp

```diff
@@ -81,18 +81,19 @@ struct in_place_shfl_op {
     union conv_type {
```

Comment:
It seems that you don't need `conv_type` anymore at all but can just do the same as some lines below for `sizeof(Scalar) == sizeof(double)`, i.e.
```
    auto bla = reinterpret_cast<shfl_type const&>(in);
    auto blabla = self().do_shfl_op(mask, bla, lane_or_delta, width);
    //------------------------------------------------
    out = reinterpret_cast<Scalar&>(blabla);
    //------------------------------------------------
```
works for me unless I am missing something.

Reply:
no this case allows to be scalar of size 3 bytes. so if I don't put the union there the underlying storage size will be off.

Reply:
What I meant to say is that your current version should be equivalent because you are never using a different type than you initialized the union with, i.e. you are only using `tmp_in` with `tmp_in.orig` and `res` through `res.conv`.

Reply:
no it is not. I reinterpret_cast tmp_in.orig as an int. That wouldn't work if I would not use a union, which guarantees that even if orig is 3 bytes the union is 4 ...